### ISSUE TITLE
Article title: The psychometric properties of Motors of COVID-19 Vaccination Acceptance Scale (MoVac-COVID19S)

### ISSUE THREAD

**Issue by ben-domingue** [ben-domingue/irw#513]
https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/U8ZYDF

**Comment by ben-domingue** [comment]
@Chia-Yi here!

**Comment by Nadiayen** [comment]
Thanks @ben-domingue!

I worked on this today, and I initially referred to some closed issues to guide me through the process. However, I’m not sure if I did everything correctly. I’d love to hear your thoughts on my work! Please let me know if anything needs revising.

Processed Data:
[imputate_(fivecountries).csv](https://github.com/user-attachments/files/17288190/imputate_.fivecountries.csv)

Code:
`df = pd.read_csv('imputate(fivecountries).tab', delimiter='\t')
df.to_csv('imputate(fivecountries).csv', index=False)
df2 = pd.read_csv('imputate(fivecountries).csv')

## Drop columns
df2.drop(columns=['group', 'Gender', 'age', 'edu_level', 'profession'], inplace=True)

## Check null value
null_values = df2.isnull().sum()
print(null_values)

new_data = []
for index, row in df2.iterrows():
    for column in df2.columns:
        new_data.append([index + 1, column, row[column]])  

op = pd.DataFrame(new_data, columns=['id', 'item', 'resp'])

## Save result to csv
op.to_csv('imputate_(fivecountries).csv', index=False)
`

**Comment by saviranadela** [comment]
hi @Nadiayen !

I reviewed the data and noticed some responses that don’t seem accurate. I checked the raw data, and they appear the same there as well. It might be best to simply remove these problematic responses?
<img width="635" alt="image" src="https://github.com/user-attachments/assets/7ba53edb-d659-48d4-b0bf-e952f334a477">

**Comment by KingArthur0205** [comment]
Hi @Nadiayen Nice work!

I believe these floating-point numbers originate from imputed data used to fill in missing responses (i.e., the column means are inserted when a participant didn’t provide an answer for that specific item).

You’ll notice that some participants’ responses are fully imputed. I suggest we make two adjustments:
1. Remove participants whose responses are entirely imputed.
2. For partially imputed responses, set the imputed values to NA and retain the remaining valid responses.

![image](https://github.com/user-attachments/assets/b2b1c647-1d4e-4f7c-8eaf-b59a8e459938)

**Comment by Nadiayen** [comment]
@saviranadela @KingArthur0205 Thank you for catching this issue and for the solutions! I didn’t notice it.
I removed the imputed responses, which totaled 45 rows. Additionally, I didn’t find any partially imputed issues.

Processed Data:
[imputate_(fivecountries).csv](https://github.com/user-attachments/files/17300306/imputate_.fivecountries.csv)

Code:
```
## Import data
df = pd.read_csv('imputate(fivecountries).tab', delimiter='\t')
df.to_csv('imputate(fivecountries).csv', index=False)
df1 = pd.read_csv('imputate(fivecountries).csv')

## Drop columns
df1.drop(columns=['group', 'Gender', 'age', 'edu_level', 'profession'], inplace=True)
df2 = df1.copy()

## Check float values
def is_float_with_nonzero_decimals(x):
    return isinstance(x, float) and not np.isclose(x, round(x), atol=1e-2)

float_rows = df2.applymap(is_float_with_nonzero_decimals).all(axis=1)
float_rows_count = float_rows.sum()
print(f"Number of rows with all float values: {float_rows_count}")

## Remove the 45 rows
df2_cleaned = df2[~float_rows]
df2_cleaned.reset_index(drop=True, inplace=True)
print(f"Shape of the DataFrame after removing rows: {df2_cleaned.shape}")

### Check null value: Not null
null_values = df2_cleaned.isnull().sum()
print(null_values)

###
new_data = []
for index, row in df2_cleaned.iterrows():
    for column in df2_cleaned.columns:
        new_data.append([index + 1, column, row[column]])  
op = pd.DataFrame(new_data, columns=['id', 'item', 'resp'])
print(op)

## Save result to csv file
op.to_csv('imputate_(fivecountries).csv', index=False)```

**Comment by KingArthur0205** [comment]
:) No problem. Ben taught me about imputation 2 weeks ago and now I get to show off ;)

**Comment by ben-domingue** [comment]
Excellent! I have added data. Note that I also added code. 

One small thing comment @Nadiayen: It helps to have a distinct name for data+code so that we can always match stuff. I have uploaded your code here:
https://github.com/ben-domingue/irw/blob/main/data/movac_pakpour2022.R
and the table wlil also be `movac_pakpour2022`. My general approach is names that are something about the scale plus the author name and year; this seems sufficiently indicative so that we shouldn't have issues with duplicates.
